### ISSUE TITLE
Server context should use import_map.json from deno.json

### ISSUE THREAD

**Issue by greym0uth** [denoland/fresh#458]
Currently the server context gets the `import_map.json` relative to the manifest's location ([code here](https://github.com/denoland/fresh/blob/10094f894728e2074e5c10b678d8328258159310/src/server/context.ts#L97)). This is an issue in cases where the manifest gets generated into the `src` directory because the `import_map.json` must also live inside the `src` directory which requires the `deno.json` to be updated to reflect that. It's not a huge issue but isn't ideal when some may expect it to be located in the project root like most projects I've seen seem to follow.

**Question**: Is it possible to get the `import_map.json` from the `deno.json` if it exists? I'm not sure if Deno exposes this as an variable, in the env, or somewhere. I did a bit of digging but couldn't find anything.

**Comment by m4rocks** [comment]
Uhm... I think it already does that. You can safely move `import_map.json` in a separate folder from `deno.json`. Check https://github.com/denoland/fresh/pull/456

**Comment by lucacasonato** [comment]
It doesn't right now (it is hardcoded to be the `import_map.json` that is co-located next to `fresh.gen.ts`. I want to change this, but there is currently an upstream Deno bug blocking this, that should be fixed in Deno 1.24.3.

**Comment by nnmrts** [comment]
In case this wasn't already thought about, please also make the whole import map path dynamic, depending on the deno config (if there is one) and the `"importMap"` setting (if there is one).

I don't call my import maps "import_map.json" because it think it looks ugly (and I know I'm in a minority here), but it would be nice if fresh would respect that.

**Comment by rojvv** [comment]
Is https://github.com/denoland/deno/issues/15432 related?

**Comment by nnmrts** [comment]
Sorry for my post and delete before, I reread the OP and felt like my comment was repetitive.

Since I'm already here, a simple workaround (I'm currently using successfully) for this issue until it's fixed:

```sh
ln -s /path/to/actual-import-map.json ./import_map.json
```

**Comment by azerum** [comment]
From the [source code](https://github.com/denoland/fresh/blob/main/src/dev/mod.ts#L106), it seems like `deno.json` is also hardcoded to be next to `fresh.gen.ts`? 

I am trying to set up monorepo with multiple fresh apps and shared backend definitions. There are common formatting and TypeScript settings in `deno.json`, but, currently, there's no way to reuse them (https://github.com/denoland/deno/issues/18132 would help) So I thought to use single `deno.json` file at root

Is it sensible idea? Should fresh not hardcode `deno.json` as well? Or is it more idiomatic to use `deno.json` per logical project?

Edit:

Having played with it, to make VSCode intellisense work, I had to separately add each project's folder to VSCode workspace. Also, I had to create symlinks to shared `import_map.json` and ended up creating separate `deno.json` files per each folder. Without having `import_map.json` in the folder, VSCode didn't pick up typings. Kind of awkward to work with monorepos as for now, maybe I am missing something simpler

I guess we should both not hardcode the file paths and allow to share common settings for `deno.json`
